src/index.ts

```diff
@@ -1,54 +1,95 @@
+const DISTRIBUTION_AMOUNT = ethers.utils.parseEther('100');
+// helper variable used to achieve a good decimal approximation in the rewards distribution calculation
+const factor = DISTRIBUTION_AMOUNT.div(100000);
 
-// This is the vote we are distributing rewards for
-const voteId = process.env.VOTE_ID
-
-// TODO: This will come from the collections contract
-const DISTRIBUTION_AMOUNT = 10000
-
+// fetch SARCO staking contract that gives stakers voting rights tokens (SarcoVR)
 const web3 = new Web3(
   new Web3.providers.HttpProvider(
     `https://${process.env.ETHEREUM_NETWORK}.infura.io/v3/${process.env.INFURA_API_KEY}`
   )
 );
+const stakingContractABI = JSON.parse(fs.readFileSync('src/abi/sarcoStaking.json'));
+const stakingContract = new web3.eth.Contract(stakingContractABI, process.env.CONTRACT_ADDRESS);
 
-const stakingContractABI  = JSON.parse(fs.readFileSync("src/abi/sarcoStaking.json"));
-const stakingContract = new web3.eth.Contract(
-  stakingContractABI,
-  process.env.CONTRACT_ADDRESS
-);
+// helper function that sums the BN values of a mapping: used to check the sum of rewards equals the initial amount distributed
+function getSum(distribution: Map<string, BigNumber>): BigNumber {
+  let sum = zero;
+  for (let i of distribution.keys()) {
+    let value: BigNumber = distribution.get(i);
+    sum = sum.add(value);
+  }
+  return sum;
+}
 
-async function main() {
-  const voteData = await fetchVoteData(web3, voteId)
-  const blockNumber = voteData.snapshotBlockNumber
- 
-  const didVoteAddresses = new Map<string, number>()
-  let totalVoteBalance = 0
+// helper function fetching the total SarcoVR held by voters for a single voteId
+// used to compute the proportions of rewards to distribute
+async function getTotalVotersBalance(_provider: any, _voteId: string | number): Promise<BigNumber> {
+  // helper function to query voting data
+  const voteData = await fetchVoteData(_provider, _voteId);
+  // snapshot blockNumber of vote
+  const blockNumber = voteData.snapshotBlockNumber;
+
+  let totalVotersBalance: BigNumber = zero;
   for (let i = 0; i < voteData.addresses.length; i++) {
-    const votingAddress = voteData.addresses[i]
-    const stakedValueAt = await stakingContract.methods.stakeValueAt(votingAddress,blockNumber).call()
-    totalVoteBalance += +stakedValueAt
-    didVoteAddresses.set(votingAddress, stakedValueAt)
+    const votingAddress = voteData.addresses[i];
+    const stakedValueAt: BigNumber = await stakingContract.methods
+      .stakeValueAt(votingAddress, blockNumber)
+      .call();
+    totalVotersBalance = totalVotersBalance.add(stakedValueAt);
   }
-  
-  const distributionMapping = new Map<string, number>();
+  return totalVotersBalance;
+}
+
+async function main() {
+  const voteData = await fetchVoteData(web3, voteId);
+  // snapshot blockNumber of DAO proposal vote
+  const blockNumber = voteData.snapshotBlockNumber;
+  // fetch total amount of SarcoVR held by voters for a given voteId
```

Comment:
Pass `voteData` as an argument for this function so that `fetchVoteData` may be called only once.